pandas/core/sparse/frame.py

```diff
@@ -191,11 +191,13 @@ def _init_spmatrix(self, data, index, columns, dtype=None,
         for col, rowvals in values.groupby(data.col):
             # get_blocks expects int32 row indices in sorted order
             rows = rowvals.index.values.astype(np.int32)
+            vals = np.array([y for x, y in sorted(rowvals.iteritems())],
```

Comment:
@kernc 
Thanks, this is indeed a faster and more elegant solution. I've committed this solution to this branch.
Just out of curiosity though, this does not seem to be fundamentally different behavior from my original solution, since they both seem to be sorting on the same index of rowvals. Am I mistaken?